client/src/pages/LessonHome/LessonHome.js

```diff
@@ -94,14 +100,10 @@ const LessonHome = ({ navigation }) => {
               item._id,
               fileType,
             ).then((value) => {
-              const updatedData = formattedVocabData.map((element) => {
-                if (element._id === formattedItem._id) {
-                  return { ...element, imageURI: value }
-                }
-                return element
-              })
-              setData(updatedData)
-              return value
+              if (mounted) {
+                formattedItem.imageURI = value
+                setData([...formattedVocabData])
```

Comment:
The reason why I had to spread the `formattedVocabData` in a list when `formattedVocabData` is already a list is because `setData` lazily checks if `formattedVocabData` is the "same" array in terms of reference. So although the `formattedItem` has been changed React still thinks the `formattedVocabData` is the same array so it doesn't trigger a rerender correctly.

Reply:
Can you add this as a comment?